sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt

```diff
@@ -167,12 +169,27 @@ internal class ViewAnnotationManagerImpl(
   private fun drawAnnotationViews(
     positionDescriptorCoreList: List<ViewAnnotationPositionDescriptor>
   ) {
-    // TODO rewrite to avoid spawning collections
-    val sortedKeyList = positionDescriptorCoreList.map { it.identifier }
-    val idsToAddSet = sortedKeyList.minus(currentViewsDrawnMap.keys)
-    val idsToDeleteSet = currentViewsDrawnMap.keys.minus(sortedKeyList)
+    idsToAddSet.clear()
+    idsToRemoveSet.clear()
+    positionDescriptorCoreList.forEach {
+      if (!currentViewsDrawnMap.keys.contains(it.identifier)) {
+        idsToAddSet.add(it.identifier)
+      }
+    }
+    currentViewsDrawnMap.keys.forEach { id ->
+      var contains = false
+      for (descriptor in positionDescriptorCoreList) {
```

Comment:
how about converting `positionDescriptorCoreList` to set and check `descriptor in positionDescriptorCoreSet`
how large this lists can be? 20, 100, 1000 items?

Reply:
Discussed offline, let's go with current solution.